### ISSUE TITLE
"bean-doctor context" broke in revision ba5032375ba8

### ISSUE THREAD

**Issue by blais** [beancount/beancount#17]
**[Original report](https://bitbucket.org/blais/beancount/issue/17) by Matthew Harris (Bitbucket: [matthew_harris](https://bitbucket.org/matthew_harris), GitHub: [mharris312](https://github.com/mharris312)).**

----------------------------------------

```
#!bash

$ cat test.beancount
2000-01-01 open Assets:Checking

$ hg update -r dfb88ced396e
$ ./setup.py install

$ bean-doctor context test.beancount 1
Hash:9ec07ee70bcbade115c2d47e57e8d301
Location: test.beancount:1


2000-01-01 open Assets:Checking                                 

$ hg update -r ba5032375ba8
$ ./setup.py install
$ bean-doctor context test.beancount 1
No entry could be found before test.beancount:1
```

**Comment by blais** [comment]
**Original comment by Martin Blais (Bitbucket: [blais](https://bitbucket.org/blais), GitHub: [blais](https://github.com/blais)).**

----------------------------------------

Awesome, thanks. I'll have a look when I get home.

**Comment by blais** [comment]
**Original comment by Martin Blais (Bitbucket: [blais](https://bitbucket.org/blais), GitHub: [blais](https://github.com/blais)).**

----------------------------------------

I can't reproduce this, works for me.

**Comment by blais** [comment]
**Original comment by Martin Blais (Bitbucket: [blais](https://bitbucket.org/blais), GitHub: [blais](https://github.com/blais)).**

----------------------------------------

Are you rebuilding between updates?

Personally I set my PYTHONPATH to my checkout and just type "make build" in the root directory.
Takes a sec.
I never ./setup.py install.

**Comment by blais** [comment]
**Original comment by Matthew Harris (Bitbucket: [matthew_harris](https://bitbucket.org/matthew_harris), GitHub: [mharris312](https://github.com/mharris312)).**

----------------------------------------

Yes, I'm rebuilding.

It's something to do with relative paths. Absolute paths to 'bean-doctor check' work fine.

It appears that in doctor.py, do_context calls load_file(filename), which passes a relative path. load_file then converts that to an absolute path, so the filename metadata contains absolute paths everywhere.

Then do_context calls render_entry_context with that relative path, which of course never matches the absolute path of any entries, so no entries are returned.

**Comment by blais** [comment]
**Original comment by Martin Blais (Bitbucket: [blais](https://bitbucket.org/blais), GitHub: [blais](https://github.com/blais)).**

----------------------------------------

Good man. Indeed. I could reproduce. Thanks, will fix.

**Comment by blais** [comment]
**Original comment by Martin Blais (Bitbucket: [blais](https://bitbucket.org/blais), GitHub: [blais](https://github.com/blais)).**

----------------------------------------

Fixed #17, bug with relative filename being used to resolve against metadata.
